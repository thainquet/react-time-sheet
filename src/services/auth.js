import axios from 'axios'

const register = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/register`, params);
}

const login = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/login`, params);
}

const changePassword = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/changePass`, params);
}

const forgotPassword = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/forgotPass`, params);
}

export default {
  login, changePassword, forgotPassword, register
}