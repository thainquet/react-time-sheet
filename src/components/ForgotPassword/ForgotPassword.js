
import React, { useState } from 'reactn'
import { authServices } from 'services'

const ForgotPass = props => {
    const { initFormParams = {
        username: '',
        email: ''
      } } = props;
    const [form, setFormInfo] = useState(initFormParams)

    const [res, resMessage] = useState('')

    const resetPass = async () => {
        resMessage(res => res = '')
        if (!form.username || !form.email) 
            resMessage(res => res = 'All fields required!')
        else {
            const params = {
                username: form.username,
                email: form.email
            }
            const reset = await authServices.forgotPassword(params)
            if (reset) resMessage(res => res = reset.data.message)  
        }      
    }

    const updateField = event => {
        setFormInfo({
            ...form,
            [event.target.name]: event.target.value
        });
    };

    return (
        <div>
            <div>
                <label htmlFor="">Username</label>
                <input type="text" name="username" autoFocus autoComplete="off" value={form.username} onChange={updateField} />
            </div>

            <div>
                <label htmlFor="">Email</label>
                <input type="text" name="email" autoComplete="off" value={form.email} onChange={updateField} />
                <br />
            </div>
            <pre>{res}</pre>
            <button onClick={() => resetPass()} id="forgot-pass-btn">Reset Password</button>
            <button onClick={() => props.history.push('/login')} id="login-btn">Login page</button>
        </div>
    )
}

export default ForgotPass;

