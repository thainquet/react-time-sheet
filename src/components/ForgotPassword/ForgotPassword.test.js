import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { authServices } from 'services';
import ForgotPassword from './ForgotPassword';

const props = {
  initFormParams: {
    username: 'thainq00',
    email: 'thainq00@gmail.com',
  },
  history: [],
  res: ''
}

it('should render correctly', () => {
  const wrapper = renderer.create(<ForgotPassword {...props} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('has 2 inputs and 2 buttons', () => {
  const wrapper = mount(<ForgotPassword {...props} />);
  expect(wrapper.find('input')).toHaveLength(2);
  expect(wrapper.find('button')).toHaveLength(2);
  wrapper.unmount();
});

it('send request for forgot password success', () => {
  const wrapper = mount(<ForgotPassword {...props} />);
  const changePasswordFunc = jest.spyOn(authServices, 'forgotPassword').mockImplementation(() => Promise.resolve({ data: { code: 200, message: 'Thanh cong' } }));
  wrapper.find('button#forgot-pass-btn').simulate('click')
  expect(changePasswordFunc).toHaveBeenCalledWith(
    props.initFormParams
  );
  wrapper.unmount();
});
