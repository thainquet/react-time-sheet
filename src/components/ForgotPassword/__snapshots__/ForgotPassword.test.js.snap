// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div>
    <label
      htmlFor=""
    >
      Username
    </label>
    <input
      autoComplete="off"
      autoFocus={true}
      name="username"
      onChange={[Function]}
      type="text"
      value="thainq00"
    />
  </div>
  <div>
    <label
      htmlFor=""
    >
      Email
    </label>
    <input
      autoComplete="off"
      name="email"
      onChange={[Function]}
      type="text"
      value="thainq00@gmail.com"
    />
    <br />
  </div>
  <pre>
    
  </pre>
  <button
    id="forgot-pass-btn"
    onClick={[Function]}
  >
    Reset Password
  </button>
  <button
    id="login-btn"
    onClick={[Function]}
  >
    Login page
  </button>
</div>
`;
